### ISSUE TITLE
Everytime warlock resummons pet, pet has different name

### ISSUE THREAD

**Issue by mistrmatt22** [gamefreedomgit/Maelstrom#4333]
**Description:**
Not really game breaking bug, but everytime warlocks pet(propably applies to DKs aswell) is summoned, he has different name.

**How to reproduce:**
Summon new Pet, either felhunter, or imp or any.

**How it should work:**
Pets should have same name

**Proof from retail:**
name of pet: pryyghon  0:40 - 
https://youtu.be/FESPBW4YMuc?t=37

 Different instance, same name of pet pryyghon 4:24 - 
https://youtu.be/FESPBW4YMuc?t=264

Same here:
Khiidom 0:40
https://youtu.be/U255BI7HVKA?t=82

Different instance, same name pet - khiidom 7:00
https://youtu.be/U255BI7HVKA?t=422

**Comment by Pornoodemon** [comment]
No. This is not a mistake.
If the demon dies or a new one is called, then its name changes. It should be so.
Warlock daemons have a "list of names", sometimes the name of a new daemon may be the same as the previous one.
But each new summon gives him a random name from the list.

**Comment by mistrmatt22** [comment]
No i disagree with you. Check this:
https://www.youtube.com/watch?v=2QYhFlPC7ig

Under player name "doomroth" he had EXACTLY same pet name(Czaatom) in like 10 different instances.

https://www.youtube.com/watch?v=7ibtrZdtJB8

Under name player name "Snutzy" in about 3 different instances always pet name "Bruukun" 

https://www.youtube.com/watch?v=NvrUrYpDauY

He has pet Droodhum in about 2 instances

at 8:25 u can barely see he is summoning new pet and its droodhum

**Comment by mistrmatt22** [comment]
> No. This is not a error/ If the demon dies or a new one is called, then its name changes,but not always. It should be so. Warlock daemons have a "list of names", sometimes the name of a new daemon may be the same as the previous one. But summon new demon gives him a random name from the list.

can you link evidence with your claim please?

**Comment by Pornoodemon** [comment]
> > Нет. Это не ошибка/ Если демон умирает или вызывается новый, то его имя меняется, но не всегда. Так и должно быть. У демонов-чернокнижников есть «список имен», иногда имя нового демона может совпадать с предыдущим. Но вызов нового демона дает ему случайное имя из списка.
> 
> можете ли вы связать доказательства с вашим утверждением, пожалуйста?

I don't remember the algorithm of how this name change works.
If I'm not mistaken, the name should change after the summon if the pet dies.

**Comment by mistrmatt22** [comment]
> > > Нет. Это не ошибка/ Если демон умирает или вызывается новый, то его имя меняется, но не всегда. Так и должно быть. У демонов-чернокнижников есть «список имен», иногда имя нового демона может совпадать с предыдущим. Но вызов нового демона дает ему случайное имя из списка.
> > 
> > 
> > можете ли вы связать доказательства с вашим утверждением, пожалуйста?
> 
> I don't remember the algorithm of how this name change works. If I'm not mistaken, the name should change after the summon if the pet dies. It is difficult for me to translate the text, so the translation may not correctly reflect the meaning of what I am writing. As far as I understand, the same name should remain with the pet until the moment he dies In any case, this does not affect the gameplay in any way. At the moment, there are many priority tasks.

thank you for your message, but you literaly gave 0 usefull information or evidence besides "you think". 

Tldr; i have given about 20 examples + i know for 100% that how it works on old AT cata and Molten wow.

if you find usefull evidence post here, thank you

**Comment by Pornoodemon** [comment]
that the names are repeated for different characters.
https://youtu.be/7ibtrZdtJB8?si=y0HFomxu1M1DNbid&t=422
Snutzy - felhunter name (Bruukun)
https://youtu.be/7ibtrZdtJB8?si=j_wA3pmuV8GWGgoE&t=835
Majinsnutz- felhunter name (Bruukun)
And in the death plan of the pet, the name is indeed preserved.
https://youtu.be/_UIiaVIwl1M?si=NII5ochFEj4Wfy_a&t=683

**Comment by UltraNix** [comment]
Warlock pets cannot be named - they are random.
```
Same here:
Khiidom 0:40
https://youtu.be/U255BI7HVKA?t=82
```
Yeah, but on the next arena match, the name of the pet is changed.

**Comment by UltraNix** [comment]
Fixed.
